api/v1alpha1/project_config_types.go

```diff
@@ -342,6 +345,76 @@ type AzureWebhookReceiverConfig struct {
 	SecretRef corev1.LocalObjectReference `json:"secretRef" protobuf:"bytes,1,opt,name=secretRef"`
 }
 
+type StaticWebhookReceiverConfig struct {
+	// SecretRef contains a reference to a Secret. For Project-scoped webhook
+	// receivers, the referenced Secret must be in the same namespace as the
+	// ProjectConfig.
+	//
+	// For cluster-scoped webhook receivers, the referenced Secret must be in the
+	// designated "cluster Secrets" namespace.
+	//
+	// The Secret's data map is expected to contain a `secret` key whose value
+	// does NOT need to be shared directly when registering a webhook.
+	// It is used only by Kargo to create a complex, hard-to-guess URL,
+	// which implicitly serves as a shared secret.
+	//
+	// +kubebuilder:validation:Required
+	SecretRef corev1.LocalObjectReference `json:"secretRef" protobuf:"bytes,1,opt,name=secretRef"`
+
+	// Rules specifies the rule that define actions to be taken when
+	// webhook events are received.
+	//
+	// +kubebuilder:validation:Required
+	Rule StaticWebhookRule `json:"rule,omitempty" protobuf:"bytes,2,opt,name=rule"`
+}
+
+// StaticWebhookAction represents an action to be performed in response to a webhook event.
+type StaticWebhookAction string
```

Comment:
You should use a `+kubebuilder:validation:Enum` marker here to constrain the allowed values.